MagazineLayout/LayoutCore/SectionModel.swift

```diff
@@ -395,13 +420,12 @@ struct SectionModel {
     }
 
     // Header frame calculation
-    if rowIndex == indexOfHeaderRow(), var headerModel = headerModel {
-      headerModel.originInSection = CGPoint(
+    if rowIndex == indexOfHeaderRow(), let existingHeaderModel = headerModel {
+      headerModel?.originInSection = CGPoint(
         x: metrics.sectionInsets.left,
         y: metrics.sectionInsets.top)
-      headerModel.size.width = metrics.width
-      updateHeaderHeight(withMetricsFrom: headerModel)
-      self.headerModel = headerModel
+      headerModel?.size.width = metrics.width
+      updateHeaderHeight(withMetricsFrom: existingHeaderModel)
```

Comment:
Avoid bug due to mutating a copy of the header model, not the header model itself.